### DISCUSSION TITLE
Can‘t Interrupting execution because Resolving Contradiction: Enabling SharedArrayBuffer and Same-Origin Policy for Pyodide Execution in Next.js

### DISCUSSION THREAD

**Discussion by Rand-Cat** [pyodide/pyodide#4047] [Q&A]
I am developing a website using Next.js, and I want to support running Python code directly in the browser. However, I have encountered a problem:

Firstly, I need to support the functionality to stop the execution of the code. Otherwise, if a user runs a code that takes a long time to execute, it may cause issues.

To support the ability to stop the code execution, the Pyodide documentation states that I can only use the SharedArrayBuffer method.

However, enabling SharedArrayBuffer requires the following security requirements:

The document needs to be in a secure context.
For top-level documents, two headers need to be set to cross-origin isolate the site:
Cross-Origin-Opener-Policy with the value "same-origin" (protects the origin from attackers)
Cross-Origin-Embedder-Policy with the value "require-corp" or "credentialless" (protects victims from the origin)
But after enabling these security measures, I cannot load pyodide = await loadPyodide({ indexURL: 'https://cdn.jsdelivr.net/pyodide/v0.23.2/full/' }); because it does not comply with the same-origin policy. Therefore, I still cannot run Python code.

It seems like I am stuck in a contradiction. How can I resolve this issue?

**Comment by hoodmane** [comment]
I think the problem is that the `indexURL` you are passing is wrong. You probably shouldn't pass `indexURL` at all, it generally works correctly without it:
```js
const py = await loadPyodide(); // no indexURL needed
```
but if you do pass it you need you include `https://`
```js
// It also works to pass a correct full indexURL
const py = await loadPyodide({indexURL: "https://cdn.jsdelivr.net/pyodide/v0.23.4/full"});
```

**Reply by Rand-Cat** [reply]
I remove the `indexURL`, but the problem is still there. then I warp my worker in to a api, it's work now. but I don't know way.
